plugins/com.google.cloud.tools.eclipse.appengine.login/src/com/google/cloud/tools/eclipse/appengine/login/GoogleLoginService.java

```diff
@@ -28,17 +44,97 @@
  */
 public class GoogleLoginService {
 
-  // TODO(chanseok): this will be needed later.
-  public static final List<String> OAUTH_SCOPES = Collections.unmodifiableList(Arrays.asList(
-      "https://www.googleapis.com/auth/userinfo#email"
+  private static final String STASH_OAUTH_CRED_KEY = "OAUTH_CRED";
+  private static final List<String> OAUTH_SCOPES = Collections.unmodifiableList(Arrays.asList(
+      "email", //$NON-NLS-1$
+      "https://www.googleapis.com/auth/cloud-platform" //$NON-NLS-1$
   ));
 
   /**
-   * Returns the credential of the active user. If there is no active user, returns {@code null}.
+   * Returns the credential of an active user (among multiple logged-in users). A login screen
+   * may be presented, e.g., if no user is logged in or login is required due to an expired
+   * credential.
+   *
+   * @param shellProvider provides a shell for the login screen if login is necessary
+   * @throws IOException can be thrown by the underlying Login API request
+   */
+  // TODO(chanseok); synchronize properly
```

Comment:
> Does this mean that at any point, only one Job will be running in Eclipse? That's cool.

No, but when you create a `Job` you can add _scheduling rules_, which can be compound rules, and the Jobs framework will interleave jobs so that no two jobs with conflicts are run at the same time.  There's support for resolving deadlocks too.
  * [Eclipse Help: Scheduling Jobs](http://help.eclipse.org/neon/index.jsp?topic=%2Forg.eclipse.platform.doc.isv%2Fguide%2Fruntime_jobs_rules.htm)
  * [On the Job: The Eclipse 3.0 Jobs API](https://eclipse.org/articles/Article-Concurrency/jobs-api.html)